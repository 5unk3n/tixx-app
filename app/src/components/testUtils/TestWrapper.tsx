import { NavigationContainer } from '@react-navigation/native'
import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import React, { PropsWithChildren } from 'react'
import { PaperProvider } from 'react-native-paper'

import { theme } from '@/theme/defaultTheme'

const queryClient = new QueryClient({
	defaultOptions: {
		queries: {
			retry: false,
			staleTime: Infinity
		}
	}
})

export const TestWrapper = ({ children }: PropsWithChildren) => {
	return (
		<PaperProvider theme={theme.dark}>
			<QueryClientProvider client={queryClient}>
				<NavigationContainer>{children}</NavigationContainer>
			</QueryClientProvider>
		</PaperProvider>
	)
}
