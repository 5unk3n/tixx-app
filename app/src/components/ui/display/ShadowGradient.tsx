import React from 'react'
import LinearGradient, {
	LinearGradientProps
} from 'react-native-linear-gradient'

import { opacityPercentToHex } from '@/utils/opacityPercentToHex'

type gradientType = '1' | '2' | '2-2' | '3' | 'main'

const colors = {
	'1': [
		`#FFFFFF00`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(20)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(40)}`,
		`#1B1A1A${opacityPercentToHex(50)}`
	],
	'2': [
		`#1B1A1A${opacityPercentToHex(40)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(20)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(15)}`,
		`#1B1A1A${opacityPercentToHex(25)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(40)}`,
		`#1B1A1A${opacityPercentToHex(50)}`,
		`#1B1A1A${opacityPercentToHex(60)}`
	],
	'2-2': [
		`#1B1A1A${opacityPercentToHex(40)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(20)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(20)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(40)}`
	],
	'3': [
		`#1B1A1A${opacityPercentToHex(40)}`,
		`#1B1A1A${opacityPercentToHex(35)}`,
		`#1B1A1A${opacityPercentToHex(25)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(5)}`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(25)}`,
		`#1B1A1A${opacityPercentToHex(35)}`,
		`#1B1A1A${opacityPercentToHex(40)}`
	],
	main: [
		`#FFFFFF00`,
		`#1B1A1A${opacityPercentToHex(10)}`,
		`#1B1A1A${opacityPercentToHex(20)}`,
		`#1B1A1A${opacityPercentToHex(30)}`,
		`#1B1A1A${opacityPercentToHex(35)}`,
		`#1B1A1A${opacityPercentToHex(50)}`,
		`#1B1A1A${opacityPercentToHex(55)}`,
		`#1B1A1A${opacityPercentToHex(60)}`
	]
} satisfies Record<gradientType, (string | number)[]>

const locations = {
	'1': [0, 0.45, 0.56, 0.67, 0.78, 0.89, 1],
	'2': [0, 0.05, 0.11, 0.17, 0.23, 0.53, 0.59, 0.65, 0.7, 0.76, 0.81, 0.91, 1],
	'2-2': [0, 0.07, 0.11, 0.17, 0.23, 0.53, 0.64, 0.76, 0.87, 1],
	'3': [0, 0.06, 0.12, 0.18, 0.23, 0.77, 0.83, 0.9, 0.95, 1],
	main: [0, 0.3, 0.44, 0.58, 0.69, 0.8, 0.9, 1]
} satisfies Record<gradientType, number[]>

interface ShadowGradientProps
	extends Omit<LinearGradientProps, 'colors' | 'locations'> {
	type: gradientType
}

export default function ShadowGradient({
	type,
	...props
}: ShadowGradientProps) {
	return (
		<LinearGradient
			{...props}
			colors={colors[type]}
			locations={locations[type]}
			className="absolute top-0 left-0 w-full h-full"
		/>
	)
}
