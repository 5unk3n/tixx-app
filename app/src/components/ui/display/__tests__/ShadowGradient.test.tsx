import { describe, it, expect, jest } from '@jest/globals'
import { render } from '@testing-library/react-native'
import React from 'react'

import { TestWrapper } from '../../../testUtils/TestWrapper'
import ShadowGradient from '../ShadowGradient'

jest.unmock('react-native-linear-gradient')

describe('ShadowGradient', () => {
	it('renders correctly for each type and matches snapshot', () => {
		const types = ['1', '2', '2-2', '3', 'main'] as const
		types.forEach((type) => {
			const { toJSON } = render(<ShadowGradient type={type} />, {
				wrapper: TestWrapper
			})
			// Create a snapshot for each type to verify correct props are passed
			expect(toJSON()).toMatchSnapshot()
		})
	})
})
