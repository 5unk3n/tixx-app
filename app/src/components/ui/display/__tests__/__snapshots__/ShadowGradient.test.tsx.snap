// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowGradient renders correctly for each type and matches snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          16777215,
          219879962,
          437983770,
          857414170,
          1293621786,
          1713052186,
          2149259802,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        [
          0,
          0.45,
          0.56,
          0.67,
          0.78,
          0.89,
          1,
        ]
      }
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`ShadowGradient renders correctly for each type and matches snapshot 2`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          1713052186,
          1293621786,
          857414170,
          437983770,
          219879962,
          219879962,
          437983770,
          639310362,
          1075517978,
          1293621786,
          1713052186,
          2149259802,
          2568690202,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        [
          0,
          0.05,
          0.11,
          0.17,
          0.23,
          0.53,
          0.59,
          0.65,
          0.7,
          0.76,
          0.81,
          0.91,
          1,
        ]
      }
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`ShadowGradient renders correctly for each type and matches snapshot 3`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          1713052186,
          1293621786,
          857414170,
          437983770,
          219879962,
          219879962,
          437983770,
          857414170,
          1293621786,
          1713052186,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        [
          0,
          0.07,
          0.11,
          0.17,
          0.23,
          0.53,
          0.64,
          0.76,
          0.87,
          1,
        ]
      }
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`ShadowGradient renders correctly for each type and matches snapshot 4`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          1713052186,
          1494948378,
          1075517978,
          437983770,
          219879962,
          219879962,
          437983770,
          1075517978,
          1494948378,
          1713052186,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        [
          0,
          0.06,
          0.12,
          0.18,
          0.23,
          0.77,
          0.83,
          0.9,
          0.95,
          1,
        ]
      }
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;

exports[`ShadowGradient renders correctly for each type and matches snapshot 5`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <BVLinearGradient
      colors={
        [
          16777215,
          437983770,
          857414170,
          1293621786,
          1494948378,
          2149259802,
          2350586394,
          2568690202,
        ]
      }
      endPoint={
        {
          "x": 0.5,
          "y": 1,
        }
      }
      locations={
        [
          0,
          0.3,
          0.44,
          0.58,
          0.69,
          0.8,
          0.9,
          1,
        ]
      }
      startPoint={
        {
          "x": 0.5,
          "y": 0,
        }
      }
      style={
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "width": "100%",
          },
          {
            "height": "100%",
          },
        ]
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
